### PR TITLE
Add JSX className support

### PR DESCRIPTION
**Pull request by filipekiss**
Add support for using `className` when using the tag surrounds in a `jsx` or `tsx` file.

### CODE REVIEW

**Comment by machakann** [comment]
Hi, thank you for your PR and sorry for my late response.
Could you explain in what kind of context the PR makes a difference?
It seems that it expands `.foo` to `className="foo"` instead of `class="foo"`, is it correct?

**Comment by filipekiss** [comment]
@machakann Correct! When writing React components, `class` is a reserved javascript keyword, so `className` is used instead. This PR detects if the files are `.jsx` (or `.tsx`, for TypeScript) and expands class as intended when doing `sat` and passing `div.myClass`, for example!

**Comment by machakann** [comment]
Thanks, I understand.

**Comment by filipekiss** [comment]
Made the requested changes and also pulled `master` branch to avoid possible conflicts :)

File: autoload/sandwich/magicchar/t.vim [hunk 1]
```diff
@@ -129,6 +129,13 @@ function! s:tokenize_custom_attributes(text, j) abort "{{{
   return [j - a:j, tokenlist]
 endfunction
 "}}}
+
+
+function! s:matches_filetype(filetype, list) abort "{{{
+    return index(a:list, a:filetype) != -1
```
Comment (machakann) [inline autoload/sandwich/magicchar/t.vim]: Excuse me I missed last time but please unify the indent to two spaces here also.

File: autoload/sandwich/magicchar/t.vim [hunk 2]
```diff
@@ -148,7 +155,11 @@ function! s:parse(tokenlist) abort  "{{{
       if item.string ==# '#'
         let i = s:handle_id(itemdict, itemlist, i)
       elseif item.string ==# '.'
-        let i = s:handle_class(itemdict, itemlist, i)
+        if s:matches_filetype(&filetype, g:sandwich#jsx_filetypes)
+            let i = s:handle_className(itemdict, itemlist, i)
```
Comment (machakann) [inline autoload/sandwich/magicchar/t.vim]: Please fix indentation into two spaces.

File: autoload/sandwich/magicchar/t.vim [hunk 3]
```diff
@@ -148,7 +155,11 @@ function! s:parse(tokenlist) abort  "{{{
...
+        else
+            let i = s:handle_class(itemdict, itemlist, i)
```
Comment (machakann) [inline autoload/sandwich/magicchar/t.vim]: Please fix indentation into two spaces.

**Comment by filipekiss** [comment]
@machakann no problem! I've reindent the file to 2 spaces. Let me know if I missed anything :)

**Comment by machakann** [comment]
Thank you for your PR and great patience!

**Pull request merged**